### ISSUE TITLE
Validator.HTTPHeaderValue changed automaticaaly

### ISSUE THREAD

**Issue by vblahh** [ESAPI/esapi-java-legacy#673]
My   Validator.HTTPHeaderValue in  validation.properties changed automatically, How  is this possible 

 Changed to **Validator.HTTPHeaderValue=^[a-zA-Z0-9()-=*.?|;,+\/:&_ %$ '-??@?????"]*$**
With the random question marks
1) Is this possible?
2)Is this expected behavior?
3)What are the reasons for it?
4) Is it possible to avoid this scenario?

Thanks in advance!

**Comment by xeno6696** [comment]
Someone on the server edited your regex.  Looks like maybe to attempt to allow an email address, but either way it would be prudent to contact your security incident response team to determine if the server was breached.  

Could be something someone in operations tried.  When I get to a computer I will see what unit tests it might break.  

But ESAPI never writes to the properties file.

**Comment by kwwall** [comment]
I agree with closing this for the reason that @xeno6696 noted, but I have 3 points to add here:

1. If you are not already doing so, change your entire application server deployments so that by default the process user id that is running as the JavaEE / servlet application server cannot write your configuration files unless you expect / desire it to. Generally a 'chmod 444' to those files will suffice. (Or for Windows, 'attrib +R' IIRC.) Indeed, make everything read-only except for the log directory / files that it writes to if you can support that. It's just good deployment hygiene.
2. You may also want to check your source code repo and make sure it wasn't updated there as well. If it was an innocent mistake, it might be updated there as well.
3. If your regex for **`Validator.HTTPHeaderValue`** was truly "^[a-zA-Z0-9()-=.?|;,+/:&_ %$ '-??@?????"]$", that would only match a single character, because there is no '*' following the character class. So the good news is it it was indeed _accidentally_ botched, it likely was done in a manner that is mostly harmless.

**Comment by xeno6696** [comment]
This appears to make the security wrapper call to getHeader fail the unit tests on line 1026 of ValidatorTest.java

So that regex appears to be ultra permissive in that NONE of our validation tests fail.  

```
    @Test
    public void testStandardHeader() {
    	Validator v = ESAPI.validator();
    	boolean isValid = v.isValidInput("HTTPHeaderValue ", "mary.poppins@gmail.com", "avaloqLooseSafeString", 2147483647, true, true);
    	assertFalse(isValid);
    }
```

This fails.

**Comment by xeno6696** [comment]
and yes, I'm making a terrible semantic mistake by using "isValid" with a boolean variable.  :-D

**Comment by kwwall** [comment]
@xeno6696 - Huh? How could that be? The entire regex that was mentioned distills down to `^[...]$` where the '...' are just a range of _individuamattchingl_ characters. And since it is anchored with '^' and '$', it doesn't seem like it should match any input longer than a single character. (Note that there is no longer a '\*' or '+' outside the character class.) So, that regex that @vblahh provided should essentially distill down to a slightly less permissive regex of '^.$'. So how can it be matching input of longer than one character? Am I missing something, or did you throw in a '*' or '+' prior to the last '$' in the regex?

**Comment by xeno6696** [comment]
@kwwall make the change in the regex in src/test/resources, pop my test into ValidatorTest, and run.

**Comment by xeno6696** [comment]
wait I goofed, one sec

**Comment by xeno6696** [comment]
```
    @Test
    public void testStandardHeader() {
    	Validator v = ESAPI.validator();
    	boolean isValid = v.isValidInput("HTTPHeaderValue ", "mary.poppins@gmail.com", "HTTPHeaderValue", 2147483647, true, true);
    	assertFalse(isValid);
    }
```

Okay, I picked the right validation rule.  (O_o)

**Comment by kwwall** [comment]
I was about to ask you what regex "avaloqLooseSafeString" was and why you were using it instead of HTTPHeaderValue. :)

**Comment by vblahh** [comment]
Thankyou @kwwall  @xeno6696  for all your help!
